### ISSUE TITLE
PHP7 compatibility issue

### ISSUE THREAD

**Issue by Ya-Ri** [w2c/ispconfig3_roundcube#53]
Hi,
it seems that php7 is incompatible with this plugin...
Can you have a look?
Regards,
YaRi

**Comment by schickel007** [comment]
I have also problems with PHP7 and this plugin. Here is the error log from roundcube:

(My systen us Ubuntu 16.04 / Apache 2.4

[28-Apr-2016 11:16:39 +0200]: PHP Error: Failed to load plugin file /var/lib/roundcube/plugins/ispconfig3_account/ispconfig3_account.php in /usr/share/roundcube/program/lib/Roundcube/r
cube_plugin_api.php on line 173 (GET /webmail/?_task=settings)                                                                                                                          
[28-Apr-2016 11:16:39 +0200]: PHP Error: Failed to load plugin file /var/lib/roundcube/plugins/ispconfig3_autoreply/ispconfig3_autoreply.php in /usr/share/roundcube/program/lib/Roundcu
be/rcube_plugin_api.php on line 173 (GET /webmail/?_task=settings)                                                                                                                      
[28-Apr-2016 11:16:39 +0200]: PHP Error: Failed to load plugin file /var/lib/roundcube/plugins/ispconfig3_autoselect/ispconfig3_autoselect.php in /usr/share/roundcube/program/lib/Round
cube/rcube_plugin_api.php on line 173 (GET /webmail/?_task=settings)                                                                                                                    
[28-Apr-2016 11:16:39 +0200]: PHP Error: Failed to load plugin file /var/lib/roundcube/plugins/ispconfig3_pass/ispconfig3_pass.php in /usr/share/roundcube/program/lib/Roundcube/rcube_p
lugin_api.php on line 173 (GET /webmail/?_task=settings)                                                                                                                                
[28-Apr-2016 11:16:39 +0200]: PHP Error: Failed to load plugin file /var/lib/roundcube/plugins/ispconfig3_spam/ispconfig3_spam.php in /usr/share/roundcube/program/lib/Roundcube/rcube_p
lugin_api.php on line 173 (GET /webmail/?_task=settings)                                                                                                                                
[28-Apr-2016 11:16:39 +0200]: PHP Error: Failed to load plugin file /var/lib/roundcube/plugins/ispconfig3_fetchmail/ispconfig3_fetchmail.php in /usr/share/roundcube/program/lib/Roundcu
be/rcube_plugin_api.php on line 173 (GET /webmail/?_task=settings)                                                                                                                      
[28-Apr-2016 11:16:39 +0200]: PHP Error: Failed to load plugin file /var/lib/roundcube/plugins/ispconfig3_filter/ispconfig3_filter.php in /usr/share/roundcube/program/lib/Roundcube/rcu
be_plugin_api.php on line 173 (GET /webmail/?_task=settings)                                                                                                                            
[28-Apr-2016 11:16:39 +0200]: PHP Error: Failed to load plugin file /var/lib/roundcube/plugins/ispconfig3_forward/ispconfig3_forward.php in /usr/share/roundcube/program/lib/Roundcube/r
cube_plugin_api.php on line 173 (GET /webmail/?_task=settings)                                                                                                                          
[28-Apr-2016 11:16:39 +0200]: PHP Error: Failed to load plugin file /var/lib/roundcube/plugins/ispconfig3_wblist/ispconfig3_wblist.php in /usr/share/roundcube/program/lib/Roundcube/rcu
be_plugin_api.php on line 173 (GET /webmail/?_task=settings)                                                                                                                            
[28-Apr-2016 11:16:39 +0200]: <36otcpsg> PHP Error: Failed to load config from /var/lib/roundcube/plugins/jqueryui/config.inc.php in /usr/share/roundcube/program/lib/Roundcube/rcube_pl
ugin.php on line 157 (GET /webmail/?_task=settings)

**Comment by christoph-kluge** [comment]
@Ya-Ri @schickel007 which version of roundcube do you run? 

I have roundcube 1.2 running with the plugins/ispconfig_* and everything works as expected having just some some "Warnings"..

**Comment by schickel007** [comment]
Hi,

I had it on ISPConfig 3.1 -> It's the beta-version on Ubuntu with PHP7. But now I switched to Debian jessie and there with ISPConfig 3.0.5.4p9 and PHP5.4 it's ok.

**Comment by christoph-kluge** [comment]
@schickel007 let me rephrase: I assume that your roundcube version does not support php7, they added php7 support in the stable release 1.2.0. 

The directories inside your log files look pretty much that you installed it via your system's package manager which might install an outdated roundcube version
> /var/lib/roundcube
> /usr/share/roundcube

**Comment by schickel007** [comment]
You're right :) It's Ubuntu with roundcube 1.2-beta from packagemanager.

On my Debian I get the stable 1.2 from the roundcube-page and do it the "nerd-way" :) It works here like expected.

But, on my other test-linux witch Arch Linux the Beta-version from roundcube was ok with PHP7 (but on Arch Linux I did not have ISPConfig nor your plugin ;) )

For me this "issue" is no issue anymore

**Comment by christoph-kluge** [comment]
I'm not the owner of that plugin :) But this might be anyway interessting for other ppl who will come across the same issues..

**Comment by ei8ht187** [comment]
Got the same issue again on Ubuntu 18.04 with nginx 1.13.8, PHP7.2 and Roundcube 1.3.6. 
Any suggestions?

**Comment by ei8ht187** [comment]
Ok, found the error. As usual Layer8... Forgot to symlink the plugins to /var/lib/roundcube/plugins
